src/point_one/fusion_engine/messages/configuration.h

```diff
@@ -391,6 +391,356 @@ struct alignas(4) CoarseOrientation {
   uint8_t reserved[2] = {0};
 };
 
+/** @} */
+
+/**************************************************************************/ /**
+ * @name Input/Output Stream Control
+ * @{
+ ******************************************************************************/
+
+/**
+ * @brief The framing protocol of a message.
+ */
+enum class ProtocolType : uint8_t {
+  INVALID = 0,
+  FUSION_ENGINE = 1,
+  NMEA = 2,
+  RTCM = 3,
+};
+
+/**
+ * @brief Get a human-friendly string name for the specified @ref
+ *        ProtocolType.
+ * @ingroup config_and_ctrl_messages
+ *
+ * @param val The enum to get the string name for.
+ *
+ * @return The corresponding string name.
+ */
+inline const char* to_string(ProtocolType val) {
+  switch (val) {
+    case ProtocolType::INVALID:
+      return "Invalid";
+    case ProtocolType::FUSION_ENGINE:
+      return "FusionEngine";
+    case ProtocolType::NMEA:
+      return "NMEA";
+    case ProtocolType::RTCM:
+      return "RTCM";
+    default:
+      return "Unrecognized";
+  }
+}
+
+/**
+ * @brief @ref ProtocolType stream operator.
+ * @ingroup config_and_ctrl_messages
+ */
+inline std::ostream& operator<<(std::ostream& stream, ProtocolType val) {
+  stream << to_string(val) << " (" << (int)val << ")";
+  return stream;
+}
+
+/**
+ * @brief Identifies a message type.
+ */
+struct alignas(4) MsgType {
+  ProtocolType protocol = ProtocolType::INVALID;
+  uint8_t reserved[1] = {0};
+  uint16_t msg_id = 0;
+};
+
+/**
+ * @brief An output rate for a message.
+ */
+struct alignas(4) MsgRate {
+  /** @brief The type of message to configure. */
+  MsgType type;
+  /**
+   * @brief Reserved value for @ref update_period_ms to indicate that the
+   *        message should come out at its max rate.
+   *
+   * Also used for messages that output at a fixed rate.
+   */
+  static constexpr uint16_t MAX_RATE = 0xFFFF;
+  /** @brief The desired message update interval (in ms). */
+  uint16_t update_period_ms = 0;
+  uint8_t reserved[2] = {0};
```

Comment:
This PR is just adding interface control, not stream control. Let's remove this message stuff for a future update.